### ISSUE TITLE
Profile of EEP: A5-10-06 seems to be wrong for setpoint temperature

### ISSUE THREAD

**Issue by nabbl** [enocean-js/enocean-js#231]
Hey there.
In the past I decoded setpoint for A5-10-06 like this:

```var temp = extractByteValue(1,255,0,0,40,dataRaw);
var setPoint = extractByteValue(2,0,255,0,40,dataRaw);

function extractByteValue(ByteNr, minByte, maxByte, minValue, maxValue, data) {
    var rawVal  = 0;
    switch(ByteNr) {
        case 1 :
    		rawVal = (data & 0xff00) >>> 8;
    		break;
    	case 2 :
    		rawVal = (data & 0xff0000) >>> 16;
    		break;
    	case 3 :
    		rawVal = (data & 0xff000000) >>> 24;
    		break;
    } 
    
    return ((maxValue-minValue)/(maxByte-minByte))*(rawVal-minByte)+minValue;
}
```

Setpoint calculation for this eep seems to be wrong because I get values like "176" instead of the real setpoint value.

**Comment by Holger-Will** [comment]
I'm sure it's working fine. 
See the fiddle here: https://jsfiddle.net/jby7fmct/20/ 
it is testing ```extractByteValue``` for all possible byte values in all possible positions...

be aware that setPoint is a relative value always scaled to 255 (see https://enocean-js.github.io/enocean-js/?eep=a5-02-01 )

which device are you using?

I have some https://www.thermokon.de/produkte/raumbediengeraete-regler/raumbediengeraete/thanos/ here. these use eep a5-10-06 as well. 
how these work: I have to set a "base" temperature and a set point range. I set them to base 20° and range 10 which means I can set the temperature between 15° to 25°.
the setPoint value in the telegram will be the value relative to the range scaled to 0/255. So if I set it to 15 the byte value will be 0 if I set it to 15 the byte value will be 255 or if I set it to 20 it will be 127...

your 176 in my case would mean i have set the temperature to 22° ( 15 + 176/255 * 10 = 22; min setpoint + bytevalue/byterange * setpointrange)

setpoint range and base value will not be sent with the telegram...
